### PR TITLE
workaround for dirft detection on label drift

### PR DESCRIPTION
**Pull request by jedrzejkozal**
Quick workaround to make drift detection work on datasets with label drift.

### CODE REVIEW

**Review by AlifAshrafee — CHANGES_REQUESTED** [review]
I have worked on a makeshift solution to the label drift problem in my latest commit. I have kept the original labels intact and later used them to identify the drifting classes for drift detection. Please take a look and let me know if it seems logical. P.S. There might be bugs as I did not spend much time testing it out.

File: utils/drift_detection.py [hunk 1]
```diff
@@ -94,3 +91,31 @@ def detect_uncertainty_drift(dataset, train_loader, model):
...
+    for cls in drifting_classes:
+        filtered_images = []
+        for img_batch, _, _, original_target_batch in train_loader:
+            mask = original_target_batch == cls
+            selected_images = img_batch[mask]
+            filtered_images.append(selected_images)
+
+        new_images = torch.cat(filtered_images, dim=0)
+
+    if new_images.size(0) > 0 and hasattr(model, 'buffer'):
```
Comment (jedrzejkozal) [inline utils/drift_detection.py]: it looks like this if should be inside for loop, but in Github it look like its not.

File: utils/training.py [hunk 2]
```diff
@@ -54,7 +54,10 @@ def evaluate(model: ContinualModel, dataset: ContinualDataset, last=False) -> Tu
         correct, correct_mask_classes, total = 0.0, 0.0, 0.0
         for data in test_loader:
             with torch.no_grad():
-                inputs, labels = data
+                if dataset.HAS_LABEL_DRIFT:
+                    inputs, labels, _ = data
```
Comment (jedrzejkozal) [inline utils/training.py]: instead of if you can write:
inputs = data[0]
labels = data[1]

and later
not_aug_inputs = data[2]
logits = data[-1]

File: utils/drift_detection.py (hunk 1 above)
Comment (AlifAshrafee) [reply utils/drift_detection.py:111]: It is actually inside the for loop in my VS Code, I don't know why it's appearing outside here. 
Screenshot 2024-08-15 134330

Comment (AlifAshrafee) [reply utils/drift_detection.py]: You are right; I changed it during this implementation. I was thinking of collecting all the fine-grained classes and then comparing them with the metaclasses from the buffer since we don't know which metaclass comes from which fine-grained class. However, if we incorporate the fine-grained labels inside the buffer, this problem will be fixed.
